tests/auxiliary/minisimd.rs

```diff
@@ -147,14 +146,80 @@ impl<T, const N: usize> PackedSimd<T, N> {
     pub const fn from_array(a: [T; N]) -> Self {
         PackedSimd(a)
     }
-    pub fn as_array(&self) -> &[T; N] {
+    pub const fn as_array(&self) -> &[T; N] {
         let p: *const Self = self;
         unsafe { &*p.cast::<[T; N]>() }
     }
-    pub fn into_array(self) -> [T; N]
+    pub const fn into_array(self) -> [T; N]
     where
         T: Copy,
     {
         *self.as_array()
     }
+    pub const fn splat(a: T) -> Self
+    where
+        T: Copy,
+    {
+        Self([a; N])
+    }
+}
+
+// As `const_trait_impl` is a language feature with specialized syntax, we have to use them in a way
+// such that it doesn't get parsed as Rust code unless `cfg(minisimd_const)` is on. The easiest way
+// for that is a macro
+
+macro_rules! impl_traits {
+    ($($const_:ident)?) => {
+        impl<T: $([$const_])? PartialEq, const N: usize> $($const_)? PartialEq for Simd<T, N> {
+            fn eq(&self, other: &Self) -> bool {
+                self.as_array() == other.as_array()
+            }
+        }
+
+        impl<T, const N: usize> $($const_)? core::ops::Index<usize> for Simd<T, N> {
+            type Output = T;
+            fn index(&self, i: usize) -> &T {
+                &self.as_array()[i]
+            }
+        }
+
+        impl<T: $([$const_])? PartialEq, const N: usize> $($const_)? PartialEq for PackedSimd<T, N>
+        {
+            fn eq(&self, other: &Self) -> bool {
+                self.as_array() == other.as_array()
+            }
+        }
+    };
 }
+
+#[cfg(minisimd_const)]
+impl_traits!(const);
+
+#[cfg(not(minisimd_const))]
+impl_traits!();
+
+/// Version of `assert_eq` that ignores fancy runtime printing in const context
```

Comment:
Nit: Could you link to the issue for const `assert_eq!`?
```suggestion
/// Version of `assert_eq` that ignores fancy runtime printing in const context.
/// FIXME: Remove once <https://github.com/rust-lang/rust/issues/119826> is fixed.
```

Reply:
I just committed from the GH UI, will squash later